server/routes/api/place.py

```diff
@@ -800,30 +801,50 @@ def api_ranking_chart(dcid):
         parent_place_dcid = EARTH_DCID
         place_type = "Country"
     else:
-        parent_place_list = get_parent_place(dcid).get(dcid, [])
-        # Get the last parent place returned.
-        if parent_place_list:
-            parent_place_dcid = parent_place_list[-1]["dcid"]
-            place_type = get_place_type(dcid)
+        place_type = get_place_type(dcid)
+        parent_place_list = parent_places(dcid).get(dcid, [])
+        for parent in parent_place_list:
+            parent_type_list = parent.get("types", [])
+            parent_place_dcid = parent.get("dcid", "")
+            if parent_type_list:
+                parent_type = parent_type_list[0]
+                # All wanted place types plus continent except CensusZipCodeTabulationArea.
+                if parent_type == "Continent" or (
+                        parent_type in ALL_WANTED_PLACE_TYPES and
+                        parent_type != "CensusZipCodeTabulationArea"):
+                    break
+        # If break is not encountered, return empty result.
         else:
             return Response(json.dumps(result),
                             200,
                             mimetype='application/json')
+
     configs = get_ranking_chart_configs()
     # Get the first stat var of each config.
     stat_vars, _ = shared_api.get_stat_vars(configs)
+    # Make sure the ranking chart configs include stat var "Count_Person".
+    if "Count_Person" not in stat_vars:
+        stat_vars.add("Count_Person")
     sv_data = dc.get_stat_set_within_place(parent_place_dcid, place_type,
                                            list(stat_vars), "")
     sv_data_values = sv_data.get("data", {})
+
     if not sv_data or not sv_data_values:
         return Response(json.dumps(result), 200, mimetype='application/json')
-    ## Get all the place names of dcids in the sv_data
+    # Get all the place names of dcids in sv_data.
     place_dcids = set()
     for sv_value in sv_data_values.values():
         sv_place_dcids = sv_value.get("stat", {}).keys()
         place_dcids = place_dcids.union(sv_place_dcids)
-    place_names = get_name(list(place_dcids))
+    place_names = get_i18n_name(list(place_dcids))
     sv_metadata = sv_data.get("metadata", {})
+    # "Count_Person" is used to filter out the places with the population less than PERSON_COUNT_LIMIT
+    places_to_rank = set()
+    count_person_data = sv_data_values.get("Count_Person", {})
+    if count_person_data:
```

Comment:
nit: this `if` statement is unnecessary because doing `.get("stat", {})` on an empty object will just return an empty object, so you can do the for-loop on line 845 without checking that count_person_data is empty or not

Reply:
Got you! Done